### PR TITLE
CI: Minor improvements

### PR DESCRIPTION
**Pull request by mobin-2008**
Hey.

ToDo:

- [x] Don't run Make based CI for Meson files changes
- [x] Don't run FreeBSD CI for Meson files changes
- [x] Upload `src/igr-tests/*/output/*` files on failure (Github actions only)
- [x] Add `DEBUG=1` flag to `make check-igr` in Make based CI

Regards

`Signed-off-by: Mobin "Hojjat" Aydinfar <mobin@mobintestserver.ir>`

### CODE REVIEW

**Comment by mobin-2008** [comment]
Looks strange, but Github actions doesn't triggered at all! Needs work (testing CI related things in fork repo is painful).

**Review by davmac314 — CHANGES_REQUESTED** [review]

File: .github/workflows/regular_ci.yml [hunk 1]
```diff
@@ -123,8 +144,8 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
-    - uses: uraimo/run-on-arch-action@v2.5.0
-      name: Getting depends & build & unit tests & integration tests
+    - name: Getting depends & build & unit tests & integration tests
```
[the remark this answers is not in the document]
Comment (mobin-2008) [reply .github/workflows/regular_ci.yml:147]: Seems like if steps include ` - uses` with `name`, cannot have another step (GitHub syntax checker gave me some weird error about it).

File: .github/workflows/meson_ci.yml [hunk 2]
```diff
@@ -17,11 +17,13 @@ on:
       - '!TODO'
       - '!LICENSE'
       - '!**/Makefile'
+      - '!.github/workflows/regular_ci.yml'
       - '!build/**'
       - '!doc/CODE-STYLE'
       - '!doc/COMPARISON'
       - '!doc/CONTRIBUTING'
       - '!doc/DESIGN'
+      - '!doc/SECURITY'
```
[the remark this answers is not in the document]
Comment (mobin-2008) [reply .github/workflows/meson_ci.yml:26]: I will test it.

File: .github/workflows/regular_ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (davmac314) [reply .github/workflows/regular_ci.yml:147]: If that was a pre-existing problem, it should be fixed in an individual commit, not squashed in with another change

[the remark this answers is not in the document]
Comment (mobin-2008) [reply .github/workflows/regular_ci.yml]: Seems like not needed

**Review by davmac314 — APPROVED** [review]
Looks much better, thanks

**Pull request merged**